src/core/thread/mle_router.cpp

```diff
@@ -111,7 +112,28 @@ void MleRouter::HandlePartitionChange(void)
 
 bool MleRouter::IsRouterEligible(void) const
 {
-    return mRouterEligible && IsFullThreadDevice();
+    bool                  rval      = true;
```

Comment:
nit: I think you could slightly simplify / reduce code size by:

1. Initializing to `false` here.
2. Removing `rval = false` from each `VerifyOrExit()` below.
3. Adding `rval = true` just before the `exit:` label.

Thoughts?

Reply:
Done, thanks for the suggestion.